### PR TITLE
Obsoleting the WinIdName header

### PR DESCRIPTION
**Pull request by andreasohlund**
As agreed in https://github.com/Particular/NServiceBus/pull/2973

I'll update doco and upgrade guide once pulled

### Related issues

Obsoletes https://github.com/Particular/NServiceBus/pull/2973
Obsoletes #2494 

v6 fix for #3057

### CODE REVIEW

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -388,6 +388,10 @@ namespace NServiceBus
         public const string SubscriptionMessageType = "SubscriptionMessageType";
         public const string TimeSent = "NServiceBus.TimeSent";
         public const string TimeToBeReceived = "NServiceBus.TimeToBeReceived";
+        [System.ObsoleteAttribute("The WinIdName header is no longer attached to outgoing message to avoid passing s" +
+            "ecurity related information on the wire. Should you rely on the header beeing pr" +
```
Comment (SimonCropp) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Beeing

File: src/NServiceBus.Core/obsoletes.cs [hunk 2]
```diff
@@ -1505,6 +1505,16 @@ public void Stop()
 }
 
 #pragma warning restore 0067
+
+namespace NServiceBus
+{
+   public static partial class Headers
```
Comment (SimonCropp) [inline src/NServiceBus.Core/obsoletes.cs]: Trying to troll my ocd with a missing space? ;)

**Comment by SimonCropp** [comment]
Apart from the typo this looks good to pull

**Comment by andreasohlund** [comment]
Thanks, had to re-install respeller into R# 10 :(

All fixed pushed now

**Pull request merged**